tests/framework/db/QueryTest.php

```diff
@@ -166,6 +167,9 @@ public function testOrder()
 
         $query->addOrderBy('age ASC, company DESC');
         $this->assertEquals(['team' => SORT_ASC, 'company' => SORT_DESC, 'age' => SORT_ASC], $query->orderBy);
+
+        $query->orderBy(new Expression('CASE WHEN x = y THEN 0 ELSE 1 END DESC'));
```

Comment:
I tried to run following test on my local machine and it passed successfully
```
$query->orderBy(new Expression('CASE WHEN x = y THEN 0 ELSE 1 END DESC, x ASC'));
$this->assertEquals(['CASE WHEN x = y THEN 0 ELSE 1 END' => SORT_DESC, 'x' => SORT_ASC], $query->orderBy);
```

I can update my PR to include this test instead previous one